src/server.rs

```diff
@@ -704,12 +730,78 @@ impl LanguageServer for LspServer {
         };
         let pkg = parse_and_analyze(contents);
         let node = find_node(
-            walk::Node::Package(&pkg),
+            SemanticNode::Package(&pkg),
             params.text_document_position.position,
         );
 
         Ok(Some(find_references(key, node)))
     }
+    async fn completion(
+        &self,
+        params: lsp::CompletionParams,
+    ) -> RpcResult<Option<lsp::CompletionResponse>> {
+        let key =
+            params.text_document_position.text_document.uri.clone();
+
+        // We need to clone here becase `params` is used later.
+        let store = self.store.lock().unwrap().clone();
+        let contents = match store.get(&key) {
+            Some(v) => v.to_string(),
+            None => {
+                error!(
+                    "textDocument/completion called on unknown file {}",
+                    key
+                );
+                return Err(lspower::jsonrpc::Error::invalid_params(
+                    format!("file not opened: {}", key),
+                ));
+            }
+        };
+
+        let items = if let Some(ctx) = params.context.clone() {
+            match (ctx.trigger_kind, ctx.trigger_character) {
+                (
+                    lsp::CompletionTriggerKind::TriggerCharacter,
+                    Some(c),
+                ) => match c.as_str() {
+                    "." => find_dot_completions(params, contents),
+                    // XXX(sean): All `find_arg_completions` does is look for bucket names
+                    // if the parameter name is "bucket". Since we don't currently
+                    // support bucket completions, this match arm is unnecessary.
+                    //
+                    // ":" => {
```

Comment:
Is this really unnecessary, or is it kind of erroneous, but not harmful to leave in? I get why it's here and commented out with a `XXX` comment, but would it be harmful to just leave it uncommented? Worst it could do is just be a no-op for every arg completion attempt right now, right?

Reply:
Sure, I don't think it'll do any harm to leave it in.

Reply:
Actually, the code that it calls requires a `RequestContext` to even function. We could create an empty context inside `find_arg_completions()` to pass into the functions that require it, and it would end up as a no-op, but if we're eventually going to delete the `RequestContext`, that means we would also have to delete this code anyway.